src/topology/unit_interval.lean

```diff
@@ -80,3 +81,52 @@ meta def unit_interval : tactic unit :=
 `[apply unit_interval.le_one] <|> `[apply unit_interval.one_minus_le_one]
 
 end tactic.interactive
+
+
+section
+variables {𝕜 : Type*} [field 𝕜] [topological_space 𝕜] [topological_ring 𝕜]
+
+/--
+The map `λ x, a * x + b`, as a homeomorphism from `𝕜` (a topological field) to itself, when `a ≠ 0`.
+-/
+@[simps]
+def affine_homeomorph (a b : 𝕜) (h : a ≠ 0) : 𝕜 ≃ₜ 𝕜 :=
```

Comment:
I don't think this belongs to this file. Can you find a better home?

Reply:
There's actually not an existing file where this can go. It needs lemmas from both `topology.algebra.group_with_zero` and `topology.algebra.ring`. I've moved it to a new file `topology.algebra.field`.